Core/Src/main.c

```diff
@@ -34,6 +35,8 @@
 
 /* Private define ------------------------------------------------------------*/
 /* USER CODE BEGIN PD */
+#define CS_PIN GPIO_PIN_4
```

Comment:
It seems like you configured GPIO PB8 as the chip select pin. The PIN4 here, which is defined as CS_PIN, might be a mistake.